core/schema/service.go

```diff
@@ -278,3 +289,87 @@ func getIDforSchema(ns, schema, dataUUID string) string {
 	key := fmt.Sprintf("%s-%s-%s", ns, schema, dataUUID)
 	return uuid.NewSHA1(uuid.NameSpaceOID, []byte(key)).String()
 }
+
+func (s *Service) DetectSchemaChange(namespace string, schemaName string, fromVersion string, toVersion string, depth string) (*stencilv1beta1.SchemaChangedEvent, error) {
+	ctx := context.Background()
+	fmt.Printf("fromVersion in request - %s", fromVersion)
+	fmt.Printf("toVersion in request- %s", toVersion)
+	fromVer, toVer, err := s.getVersions(ctx, namespace, schemaName, fromVersion, toVersion)
+	if err != nil {
+		return nil, err
+	}
+	fmt.Printf("fromVersion after conversion - %d", fromVer)
+	fmt.Printf("toVersion after conversion- %d", toVer)
+	if depth == "" {
+		depth = strconv.Itoa(-1)
+	}
+	depth64, errDepth := strconv.ParseInt(depth, 10, 32)
+	if errDepth != nil {
+		return nil, fmt.Errorf("invalid depth")
+	}
+	_, fromVerData, fromVerDataError := s.Get(ctx, namespace, schemaName, fromVer)
+	if fromVerDataError != nil {
+		return nil, fmt.Errorf("error getting data for version %d - %s", fromVer, fromVerDataError.Error())
+	}
+	_, toVerData, toVerDataError := s.Get(ctx, namespace, schemaName, toVer)
+	if toVerDataError != nil {
+		return nil, fmt.Errorf("error getting data for version %d - %s", toVer, toVerDataError.Error())
+	}
+	req := &changedetector.ChangeRequest{
+		NamespaceID: namespace,
+		SchemaName:  schemaName,
+		OldData:     fromVerData,
+		NewData:     toVerData,
+		Version:     toVer,
+		Depth:       int32(depth64),
+	}
+	sce, err := s.changeDetectorService.IdentifySchemaChange(ctx, req)
+	if err != nil {
+		return sce, fmt.Errorf("got error while identifying schema change for namespace : %s, schema: %s, version: %d, %s", req.NamespaceID, req.SchemaName, req.Version, err.Error())
+	}
+	if err := s.sendNotification(ctx, sce, req, -1); err != nil {
```

Comment:
what is -1 in this call?

Reply:
We need to pass schemaID in sendNotification method. If we are not passing schemaID or passing it as -1, then the sendNotification method will fetch it from the database.